tests/MongoDB.Driver.Tests/OcspIntegrationTests.cs

```diff
@@ -0,0 +1,144 @@
+             * the tlsInsecure version.  */
+            var secureClientException = Record.Exception(() => Ping());
+            var tlsInsecureClientException = Record.Exception(()=> Ping(tlsInsecure: true));
+
+            tlsInsecureClientException.Should().BeNull();
+            if (secureClientShouldConnect)
+            {
+                secureClientException.Should().BeNull();
+            }
+            else
+            {
+                secureClientException.Should().BeOfType<TimeoutException>();
+                var message = secureClientException.Message;
+                // The exception will lack this message if the heartbeat doesn't fire
+                message.Should().Contain("The remote certificate is invalid according to the validation procedure");
+            }
+
+            void Ping(bool tlsInsecure = false)
+            {
+                using (var client = CreateDisposableMongoClient(tlsInsecure))
+                {
+                    client.GetDatabase("admin").RunCommand<BsonDocument>(new BsonDocument("ping", 1));
+                    if (client.Settings.SdamLogFilename != null)
+                    { // Log file needs a bit of time to be written before we dispose the client
+                        System.Threading.Thread.Sleep(2000);
+                    }
+                }
+            }
+        }
+
+        // private methods
+        /* We can't use DriverTestConfiguration.CreateDisposableClient because that results in a call to check
+         * the cluster type, which fails because a connection cannot be established to the mongod when testing
+         * the revoked certificate case */
+        private DisposableMongoClient CreateDisposableMongoClient(bool tlsInsecure)
+        {
+            var settings = DriverTestConfiguration.GetClientSettings().Clone();
+            settings.SslSettings.CheckCertificateRevocation = true;
+            // setting AllowInsecureTls= true will automatically set CheckCertificateRevocation to false
+            settings.AllowInsecureTls = tlsInsecure;
+            /* We want the heartbeat to fire so that we can get the HeartBeat exception in the cluster description
+             * in the exception that will be thrown when testing invalid certificates */
+            settings.HeartbeatInterval = TimeSpan.FromMilliseconds(500);
+            /* We lower the server selection timeout to speed up tests, but choosing a value <=5s may
+             * result in the driver being unable to perform OCSP endpoint checking in time, causing a
+             * ServerSelectionTimeout that does include a certificate revocation status error message. */
+            settings.ServerSelectionTimeout = TimeSpan.FromSeconds(5*2); // must be > 5s
+            // settings.SdamLogFilename = @"C:\temp\sdam" + $"{tlsInsecure}.log";
+
+            return new DisposableMongoClient(new MongoClient(settings));
+        }
+
+        private bool GetSecureClientExpectedOutcomeOrSkipTest()
```

Comment:
I would split this into two methods:

    private bool GetShouldSucceed()
    {
        var environmentVariableValue = Environment.GetEnvironmentVariable(_shouldSucceedEnvironmentVariableName);
        if (!Boolean.TryParse(environmentVariableValue, out var shouldSucceed))
        {
            throw new FormatException(
                $"Invalid value for {_shouldSucceedEnvironmentVariableName} environment variable: {environmentVariableValue}. Expected true or false.");
        }
        return shouldSucceed;
    }

    private void SkipTestIfShouldSucceedEnvironmentVariableIsNotSet()
    {
        if (Environment.GetEnvironmentVariable(_shouldSucceedEnvironmentVariableName) == null)
        {
            throw new SkipException($"Skipping because {_shouldSucceedEnvironmentVariableName} not set.");
        }
    }